### PR TITLE
Fix left right comparison

### PR DESCRIPTION
**Pull request by hugoboos**

### CODE REVIEW

File: what-changed/model/operation_test.go [hunk 1]
```diff
@@ -1310,6 +1311,62 @@ security:
...
+	_ = low.BuildModel(lNode.Content[0], &lDoc)
+	_ = low.BuildModel(rNode.Content[0], &rDoc)
+	_ = lDoc.Build(lNode.Content[0], nil)
+	_ = rDoc.Build(rNode.Content[0], nil)
+
+	// compare.
+	extChanges := CompareOperations(&lDoc, &rDoc)
+	assert.Equal(t, 1, extChanges.TotalChanges())
+	assert.Equal(t, 0, extChanges.TotalBreakingChanges())
+	assert.Empty(t, extChanges.SecurityRequirementChanges)
```
Comment (hugoboos) [inline what-changed/model/operation_test.go:1339]: This assert tests the current behavior. Not sure is this is desired.

File: what-changed/model/operation.go [hunk 2]
```diff
@@ -4,526 +4,527 @@
...
+				nil)
+		}
+		if lParamsUntyped.IsEmpty() && !rParamsUntyped.IsEmpty() {
+			CreateChange(&changes, PropertyAdded, v3.ParametersLabel,
+				nil, rParamsUntyped.ValueNode, true, nil,
+				rParamsUntyped.Value)
+		}
+
+		// security
+		if !lOperation.Security.IsEmpty() || !rOperation.Security.IsEmpty() {
```
Comment (hugoboos) [inline what-changed/model/operation.go:336]: Should apply for the other fields as well?

**Review by daveshanley — COMMENTED** [review]
I am unable to see what has changed here, too much of the code has been shifted using formatting. I don't know what these changes are?

**Comment by hugoboos** [comment]
The most changes are due to whitespace. You can toggle them at the overview.

The fixes are for lines like these:  `if len(lDoc.Schemes.Value) > 0 || len(lDoc.Schemes.Value) > 0`. Where the `lDoc` is checked two times.

File: what-changed/model/operation.go (hunk 2 above)
Comment (daveshanley) [reply what-changed/model/operation.go:336]: Thank you for finding this!

Comment (daveshanley) [reply what-changed/model/operation.go:336]: There are a few cases where these copy/paste may occur, if you find any more, the same treatment should be applied yes.

File: what-changed/model/path_item.go [hunk 3]
```diff
@@ -4,595 +4,596 @@
...
+		CreateChange(changes, PropertyRemoved, v3.PatchLabel,
+			lPath.Patch.ValueNode, nil, true, lPath.Patch.Value, nil)
+	}
+	if lPath.Patch.IsEmpty() && !rPath.Patch.IsEmpty() {
+		CreateChange(changes, PropertyAdded, v3.PatchLabel,
+			nil, rPath.Patch.ValueNode, false, nil, lPath.Patch.Value)
+	}
+
+	// parameters
+	if !lPath.Parameters.IsEmpty() && !rPath.Parameters.IsEmpty() {
```
Comment (daveshanley) [inline what-changed/model/path_item.go:286]: Thanks for fixing.

File: what-changed/model/operation_test.go (hunk 1 above)
Comment (daveshanley) [reply what-changed/model/operation_test.go:1339]: This looks correct to me.

**Review by daveshanley — APPROVED** [review]
Thanks for fixing these issues! I really appreciate your contribution.

File: what-changed/model/operation.go (hunk 2 above)
Comment (hugoboos) [reply what-changed/model/operation.go:336]: To be sure we mean the same: `&&` -> `||`?

**Comment by hugoboos** [comment]
I've split the whitespace from the actual fix for clarity.

**Comment by hugoboos** [comment]
Might be good to correct the indentation of all the files at once.

**Review by daveshanley — APPROVED** [review]
Thank you for your contribution, and for making `libopenapi` a better library.

**Pull request merged**